lib/msal-browser/src/network/XhrClient.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { INetworkModule, NetworkRequestOptions } from "msal-common";
+
+export class XhrClient implements INetworkModule {
+
+    async sendGetRequestAsync<T>(url: string, options?: NetworkRequestOptions): Promise<T> {
+        return this.sendRequestAsync(url, "GET", options);
+    }    
+    
+    async sendPostRequestAsync<T>(url: string, options?: NetworkRequestOptions): Promise<T> {
+        return this.sendRequestAsync(url, "POST", options);
+    }
+
+    private sendRequestAsync<T>(url: string, method: string, options?: NetworkRequestOptions): Promise<T> {
+        return new Promise<T>((resolve, reject) => {
+            const xhr = new XMLHttpRequest();
+            xhr.open(method, url, /* async: */ true);
+            this.setXhrHeaders(xhr, options.headers);
+            xhr.onload = (ev) => {
+                if (xhr.status < 200 || xhr.status >= 300) {
+                    reject(this.handleError(xhr.responseText));
+                }
+                let jsonResponse: T;
+                try {
+                    jsonResponse = JSON.parse(xhr.responseText) as T;
+                } catch (e) {
+                    reject(this.handleError(xhr.responseText));
+                }
+
+                resolve(jsonResponse);
```

Comment:
I think you can put this inside the try, then that way you can put line 26 inside the try as well.

```js
try {
  const jsonResponse = JSON.parse(xhr.responseText) as t;
  return jsonResponse;
} catch (e) {
  reject(this.handleError(xhr.responseText));
}
```

Reply:
done